internal/statestore/redis.go

```diff
@@ -637,3 +639,112 @@ func (rb *redisBackend) newExponentialBackoffStrategy() backoff.BackOff {
+	_, err = redisConn.Do("SET", backfill.GetId(), value)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to set the value for backfill, id: %s", backfill.GetId())
+		return status.Errorf(codes.Internal, "%v", err)
+	}
+
+	return nil
+}
+
+// GetBackfill gets the Backfill with the specified id from state storage. This method fails if the Backfill does not exist.
+func (rb *redisBackend) GetBackfill(ctx context.Context, id string) (backfill *pb.Backfill, ticketIds []string, err error) {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return nil, nil, status.Errorf(codes.Unavailable, "GetBackfill, id: %s, failed to connect to redis: %v", id, err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	value, err := redis.Bytes(redisConn.Do("GET", id))
+	if err != nil {
+		// Return NotFound if redigo did not find the ticket in storage.
+		if err == redis.ErrNil {
+			msg := fmt.Sprintf("Ticket id: %s not found", id)
+			return nil, nil, status.Error(codes.NotFound, msg)
+		}
+
+		err = errors.Wrapf(err, "failed to get the ticket from state storage, id: %s", id)
+		return nil, nil, status.Errorf(codes.Internal, "%v", err)
+	}
+
+	if value == nil {
+		msg := fmt.Sprintf("Ticket id: %s not found", id)
+		return nil, nil, status.Error(codes.NotFound, msg)
+	}
+
+	backfill = &pb.Backfill{}
+	if err = proto.Unmarshal(value, backfill); err != nil {
+		err = errors.Wrapf(err, "failed to unmarshal the backfill proto, id: %s", id)
+		return nil, nil, status.Errorf(codes.Internal, "%v", err)
+	}
+
+	return backfill, nil, nil
+}
+
+// DeleteBackfill removes the Backfill with the specified id from state storage. This method succeeds if the Backfill does not exist.
+func (rb *redisBackend) DeleteBackfill(ctx context.Context, id string) error {
+	return nil
+}
+
+// UpdateBackfill updates an existing Backfill with a new data. Caller has to provide a custom updateFunc if this function is called not for the game server.
+func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfill, ticketIds []string) error {
+	return nil
+}
+
+// IndexBackfill adds the backfill to the index.
+func (rb *redisBackend) IndexBackfill(ctx context.Context, backfill *pb.Backfill) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return status.Errorf(codes.Unavailable, "IndexBackfill, id: %s, failed to connect to redis: %v", backfill.GetId(), err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	err = redisConn.Send("SADD", allBackfills, backfill.Id)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to add backfill to all backfills set, id: %s", backfill.Id)
+		return status.Errorf(codes.Internal, "%v", err)
+	}
+
+	return nil
+}
+
+// DeindexBackfill removes specified Backfill from the index. The Backfill continues to exist.
+func (rb *redisBackend) DeindexBackfill(ctx context.Context, id string) error {
+	return nil
+}
+
+func (rb *redisBackend) GetIndexedBackfills(ctx context.Context) (map[string]struct{}, error) {
```

Comment:
This function should return pairs: BackfillIds, and Versions.
Can we move Index to another PR?

Reply:
Already implemented